testnet/stacks-node/src/nakamoto_node/relayer.rs

```diff
@@ -852,104 +877,219 @@ impl RelayerThread {
         Ok(())
     }
 
-    fn continue_tenure(&mut self, new_burn_view: ConsensusHash) -> Result<(), NakamotoNodeError> {
-        if let Err(e) = self.stop_tenure() {
-            error!("Relayer: Failed to stop tenure: {e:?}");
-            return Ok(());
+    /// Get a snapshot for an existing burn chain block given its consensus hash.
+    fn get_block_snapshot_consensus(
+        &self,
+        ch: &ConsensusHash,
+    ) -> Result<BlockSnapshot, NakamotoNodeError> {
+        SortitionDB::get_block_snapshot_consensus(self.sortdb.conn(), ch)
+            .map_err(|e| {
+                error!(
+                    "Relayer: failed to get block snapshot for new burn view: {:?}",
+                    e
+                );
+                NakamotoNodeError::SnapshotNotFoundForChainTip
+            })?
+            .ok_or_else(|| {
+                error!("Relayer: failed to get block snapshot for new burn view");
+                NakamotoNodeError::SnapshotNotFoundForChainTip
+            })
+    }
+
+    /// Get the Stacks block ID for the canonical tip.
+    fn get_canonical_stacks_tip(&self) -> StacksBlockId {
```

Comment:
Can you instead add this to `SortitionDB` as a public method?

Reply:
Also maybe rename to `get_canonical_stacks_tip_block_id`, though I'm not really confident on that suggestion. I think my suggestion would follow other functions like `SortitionDB::get_canonical_stacks_chain_tip_hash_and_height`

Reply:
Added in c2131150e636190c756e6e695d44a544dd419daa

Reply:
Thanks!